jar-persistence/src/test/java/com/sixsq/slipstream/persistence/VersionCounterTest.java

```diff
@@ -22,14 +22,53 @@
 
 import static org.hamcrest.Matchers.is;
 import static org.junit.Assert.assertThat;
-
 import org.junit.Test;
 
+import java.util.ArrayList;
+import java.util.List;
+import java.util.logging.Logger;
+
 public class VersionCounterTest {
 
+	private static final int numberOfThreads = 30;
+
 	@Test
 	public void increment() {
 		int counter = VersionCounter.getNextVersion();
+
 		assertThat(VersionCounter.getNextVersion(), is(counter+1));
+		assertThat(VersionCounter.getNextVersion(), is(counter+2));
+	}
+
+	@Test
+	public void testParallelIncrement() throws InterruptedException {
+		List<Thread> threads = new ArrayList<>();
+
+		int initialVersion = VersionCounter.getNextVersion();
+		int finalVersion = initialVersion + numberOfThreads + 1;
```

Comment:
Please set `finalVersion` as `final`, or move the sum directly to the assertion (line 63). In the latter case, define `initialVersion` as `final` as well.